### PR TITLE
[Templates]: Add QR Code Generator to Examples

### PR DESCRIPTION
**Pull request by lauragift21**
Fixes # [insert GH or internal issue number(s)].

**What this PR solves / how to test:**
Add a QR Code Source code to examples directory

**Associated docs issue(s)/PR(s):**

- https://github.com/cloudflare/cloudflare-docs/pull/9414

**Author has included the following, where applicable:**

- [ ] Tests
- [ ] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

File: templates/examples/qr-code-generator/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"name": "qr-code-generator",
+	"version": "0.0.0",
+	"private": true,
+	"scripts": {
+		"deploy": "wrangler publish",
```
Comment (Cherry) [inline templates/examples/qr-code-generator/package.json:6]:
```suggestion
		"deploy": "wrangler deploy",
```

As per https://github.com/cloudflare/workers-sdk/pull/3486, it'd make sense to align this with the rest of the templates using `deploy` over `publish` now.

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
I would prefer if the newer templates were in TypeScript. It is easier to downgrade to JS then it is to setup a TS environment from scratch.

File: templates/examples/qr-code-generator/package.json (hunk 1 above)
Comment (lauragift21) [reply templates/examples/qr-code-generator/package.json:6]: Thanks @Cherry!

**Comment by lauragift21** [comment]
> I would prefer if the newer templates were in TypeScript. It is easier to downgrade to JS then it is to setup a TS environment from scratch.

Thanks, @JacobMGEvans! This is an example from one of the tutorials, rather than a template. but I've gone ahead to update it to a TS version.

**Comment by JacobMGEvans** [comment]
can you add a `tsconfig.json`

```json
{
	"compilerOptions": {
		"noEmit": true,
		"module": "esnext",
		"target": "esnext",
		"lib": ["esnext"],
		"strict": true,
		"moduleResolution": "node",
		"types": ["@cloudflare/workers-types"]
	},
	"exclude": ["node_modules", "dist", "test"]
}
```

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
Out of interest I tried this out with (rather than `node_compat`) the `nodejs_compat` and `experimental:nodejs_compat_v2` flags.

It totally failed to build with `nodejs_compat`: 

```
✘ [ERROR] Could not resolve "zlib"

    node_modules/qr-image/lib/png.js:3:19:
      3 │ var zlib = require('zlib');
        ╵                    ~~~~~~

  The package "zlib" wasn't found on the file system but is built into node.
  Add "node_compat = true" to your wrangler.toml file and make sure to prefix the module name with "node:" to enable Node.js compatibility.


✘ [ERROR] Could not resolve "stream"

    node_modules/qr-image/lib/qr.js:3:23:
      3 │ var Readable = require('stream').Readable;
        ╵                        ~~~~~~~~
```

With `experimental:nodejs_compat_v2` it built but then failed at runtime:

```
[wrangler:err] Error: [unenv] zlib.deflateSync is not implemented yet!
    at createNotImplementedError (node_modules/.pnpm/unenv-nightly@1.10.0-1717606461.a117952/node_modules/unenv-nightly/runtime/_internal/utils.mjs:22:9)
    at Object.fn2 [as deflateSync] (node_modules/.pnpm/unenv-nightly@1.10.0-1717606461.a117952/node_modules/unenv-nightly/runtime/_internal/utils.mjs:26:11)
    at Object.png (templates/examples/qr-code-generator/node_modules/qr-image/lib/png.js:23:14)
    at Object.qr_image_sync [as imageSync] (templates/examples/qr-code-generator/node_modules/qr-image/lib/qr.js:102:13)
    at generateQRCode (templates/examples/qr-code-generator/src/worker.ts:20:20)
```

Something to fix before we go GA with nodejs_v2 @IgorMinar, @jculvey and @jasnell ?

**Pull request merged**